python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -974,5 +944,41 @@ def applyModelWeights(self, modelImages, refImage, modelWeights):
         """
         if self.config.useModelWeights:
             for model in modelImages:
-                model.image.array *= modelWeights
-                model.image.array += refImage.image.array*(1. - modelWeights)/self.config.dcrNumSubfilters
+                model.array *= modelWeights
+                model.array += refImage.array*(1. - modelWeights)/self.config.dcrNumSubfilters
+
+    def loadSubExposures(self, bbox, statsCtrl, tempExpRefList, imageScalerList, spanSetMaskList):
+        """Pre-load sub-regions of a list of exposures.
+
+        Parameters
+        ----------
+        bbox : `lsst.afw.geom.box.Box2I`
+            Sub-region to coadd
+        statsCtrl : `lsst.afw.math.StatisticsControl`
+            Statistics control object for coadd
+        tempExpRefList : `list` of `lsst.daf.persistence.ButlerDataRef`
+            The data references to the input warped exposures.
+        imageScalerList : `list` of `lsst.pipe.task.ImageScaler`
+            The image scalars correct for the zero point of the exposures.
+        spanSetMaskList : `list` of `dict` containing spanSet lists, or None
+            Each element is dict with keys = mask plane name to add the spans to
+
+        Returns
+        -------
+        subExposures : `dict` of `lsst.afw.image.ExposureF`
+            The pre-loaded exposures for the current subregion.
+        """
+        tempExpName = self.getTempExpDatasetName(self.warpType)
+        zipIterables = zip(tempExpRefList, imageScalerList, spanSetMaskList)
+        subExposures = {}
+        for tempExpRef, imageScaler, altMaskSpans in zipIterables:
+            exposure = tempExpRef.get(tempExpName + "_sub", bbox=bbox)
+            if altMaskSpans is not None:
+                self.applyAltMaskPlanes(exposure.mask, altMaskSpans)
+            imageScaler.scaleMaskedImage(exposure.maskedImage)
+            # Note that the variance plane here is used to store weights, not the actual variance
+            exposure.variance.array[:, :] = 0.
+            exposure.variance.array[exposure.mask.array & statsCtrl.getAndMask() == 0] = 1.
+            exposure.image.array[exposure.mask.array & statsCtrl.getAndMask() > 0] = 0.
```

Comment:
Why are these treated differently? You use `exposure.mask` to set both, but the opposite statsCtrl masks.

Reply:
It's setting the weight of unmasked pixels to 1., and the value of masked pixels to 0. I've added parenthesis to make the order of operations clear, and a comment to explain what I'm doing to the two arrays.